packages/core/src/modules/attendance/attendee-repository.ts

```diff
@@ -24,24 +29,32 @@ export interface AttendeeRepository {
   getFirstUnreservedByAttendancePoolId(id: AttendancePoolId): Promise<Attendee | null>
   getByUserId(userId: UserId, attendanceId: AttendanceId): Promise<Attendee | null>
   poolHasAttendees(poolId: AttendancePoolId): Promise<boolean>
-  checkCapacityAndReserve(attendeeId: AttendeeId): Promise<void>
+  reserveAttendee(attendeeId: AttendeeId): Promise<Attendee>
   moveFromMultiplePoolsToPool(oldPoolIds: AttendancePoolId[], newPoolId: AttendancePoolId): Promise<void>
   removeAllSelectionResponsesForSelection(attendanceId: AttendanceId, selectionId: string): Promise<void>
 }
 
 export class AttendeeRepositoryImpl implements AttendeeRepository {
   private readonly db: DBClient
+  private readonly includeUserFlags = {
+    user: {
+      select: {
+        flags: true,
+      },
+    },
+  } satisfies Prisma.AttendeeInclude
```

Comment:
Make this a const at file level. Properties should be state related to the class instance